### PR TITLE
[CHERRY-PICK] MinPlatformPkg: Add Mem Type Info variable validity checks

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Added additional logic for checking if the memory bucket information is sane.  If it's not use the default memory buckets.  This was updated to pull the upstream change instead of using the original change.

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested with a bad memory bucket configuration that caused a hang during boot.  With this change the input is correctly checked and set to the default values allowing the platform to boot.

## Integration Instructions

N/A

### CODE REVIEW

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c [hunk 1]
```diff
@@ -186,6 +187,29 @@ BuildMemoryTypeInformation (
                                &DataSize,
                                &MemoryData
                                );
+
+  if (!EFI_ERROR(Status)) {
```
Comment (makubacki) [inline MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c]: While we're not running Uncrustify here, please add a space before opening parentheses throughout the code additions.

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c
Comment (makubacki) [inline MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c:1]: You'll need `MU_CHANGE` tags throughout the changes in this file.

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c [hunk 2]
```diff
@@ -186,6 +187,29 @@ BuildMemoryTypeInformation (
                                &DataSize,
                                &MemoryData
                                );
+
```
Comment (makubacki) [inline MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c]: Can you please add an `EFI_ERROR()` check for the status from `PeiServicesLocatePpi()` and return (in addition to the assert)?

Comment (makubacki) [reply MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c]: Note: Had to add a comment about it here due to GitHub UI limitations around leaving comments outside direct code changes.

**Review by makubacki — APPROVED** [review]

**Comment by makubacki** [comment]
@kenlautner, since this came from pre-existing code, I went ahead and sent it to the list here - https://edk2.groups.io/g/devel/message/106670. If that is able to get merged tomorrow (Thursday), let's update this to be a cherry-pick. Otherwise, it is not a problem to make the PR updates requested and get this in with `MU_CHANGE` tags.

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c (hunk 1 above)
Comment (kenlautner) [reply MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c]: Updated.

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c (hunk 2 above)
Comment (kenlautner) [reply MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c]: Added.

File: MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c
Comment (kenlautner) [reply MinPlatformPkg/PlatformInit/PlatformInitPei/PlatformInitPreMem.c:1]: Updated.

**Review by srisrid — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

**Pull request merged**